src/braket/circuits/synthesis/one_qubit_decomposition.py

```diff
@@ -0,0 +1,215 @@
+# Copyright Amazon.com, Inc. or its affiliates. All Rights Reserved.
+#
+# Licensed under the Apache License, Version 2.0 (the "License"). You
+# may not use this file except in compliance with the License. A copy of
+# the License is located at
+#
+#     http://aws.amazon.com/apache2.0/
+#
+# or in the "license" file accompanying this file. This file is
+# distributed on an "AS IS" BASIS, WITHOUT WARRANTIES OR CONDITIONS OF
+# ANY KIND, either express or implied. See the License for the specific
+# language governing permissions and limitations under the License.
+
+import numpy as np
+import braket.circuits as braket_circ
+
+from braket.circuits.synthesis.util import to_su
+from braket.circuits.quantum_operator_helpers import is_unitary, eq_up_to_phase
+from braket.circuits import Circuit
+from braket.circuits.gates import X, Y, Z, Rx, Ry, Rz
+from braket.circuits.instruction import Instruction
+
+from scipy.linalg import expm
+
+import time
+
+y = Y().to_matrix()
+z = Z().to_matrix()
+
+
+class OneQubitDecomposition:
```

Comment:
Mostly to make it consistent with two_qubit_decomposition and multi_qubit_decomposition.

Reply:
I'm starting to think that it makes more sense for each of these decompositions to be functions rather than classes; reuse is easy enough by storing the output of the decomposition. For example,
```python
def one_qubit_decomposition(unitary: np.ndarray, ...) -> Circuit:
    ...
    return circ
```
After all, it makes more sense to think of decomposing a unitary as a verb rather than a decomposition as a noun.